### DISCUSSION TITLE
In-app purchases (IAP)?

### DISCUSSION THREAD

**Discussion by dfabulich** [skiptools/skip#196] [Q&A] [answered]
How would a Skip app handle in-app purchases? I couldn't find any information about this on https://skip.tools/

**Comment by marcprux** [comment, marked as answer]
We don't (yet) have any framework that abstracts the in-app purchase functionality on iOS (StoreKit) and Android (Google Play Billing Library). This would be a great addition to the Skip ecosystem, and we're hoping that a community effort will get underway to implement this (it would probably be similar to Flutter's high-level [in_app_purchase](https://pub.dev/documentation/in_app_purchase/latest/in_app_purchase/in_app_purchase-library.html) library, which delegates to the respective low-level [store_kit_wrappers](https://pub.dev/documentation/in_app_purchase_storekit/latest/store_kit_wrappers/store_kit_wrappers-library.html) and [billing_client_wrappers](https://pub.dev/documentation/in_app_purchase_android/latest/billing_client_wrappers/billing_client_wrappers-library.html) libraries).

In the meantime, you can implement this yourself by branching your platform-specific logic inside `#if SKIP` blocks, as per our [Platform Customization](https://skip.tools/docs/platformcustomization/) guide. We'd be happy to help you along with this process, and be even happier to collaborate on a community project to create a general-purpose Skip framework that handles in-app purchase!

**Comment by ptrkstr** [comment]
@marcprux even a step before implementing the native In App Purchase API, I would suggest wrapping the [Revenue Cat](https://www.revenuecat.com/) SDK in the same way that SkipFirebase has been, given Revenue Cat is the leading In App Library for iOS and Android and the APIs are the same (I believe). I personally use it and find the abstraction on top of In App Purchases very easy to use.
